requests-for-comments/2022-11-07-inserter-service.md

```diff
@@ -0,0 +1,76 @@
+# Inserter Service Requirements
+
+We’ve used the `kafka` table engine on ClickHouse for some time now. It was the perfect table engine to support our ingestion pipeline at the time. We had very few resources and needed a way to safely get events from our `[event.py](https://github.com/PostHog/posthog/blob/master/posthog/api/event.py)` ingestion service into ClickHouse with very little fuss.
+
+It’s time now for us to move on from our friend the `kafka` table engine and move this functionality outside of ClickHouse.
+
+## Why?
+
+- [clickhouse.cloud](http://clickhouse.cloud) does not support `kafka` table engines
+- It’s generally well known that the `kafka` table engine is buggy. We’ve seen this a few times with ClickHouse trying to aggressively to retry and pull down Kafka. Altinity and ClickHouse Inc have warned us against using it.
+- We’d like to decouple our pipeline from ClickHouse.
+- We’d like to be able to insert rows into multiple ClickHouse warehouses (by team) without an explosion of topics.
+- We’d like to support dynamic and parametrizable warehouse connection parameters
+  - Team 2 → Warehouse with connection details x.y.z pulled from API
+- Use whatever serialization we want on ClickHouse without having to ship the schema as a file to ClickHouse and reboot it (Proto and Avro problems)
+- Much more configurable error handling.
+- We can instrument much better metrics on the health of this step in our pipeline.
+
+# Goals
```

Comment:
This consumer should be pausable within (async) migrations. Otherwise a whole category of options we have for migrations becomes unavailable.